src/rust/runtime/scheduler/scheduler.rs

```diff
@@ -229,13 +199,15 @@ impl Scheduler {
         (waker_page_index << WAKER_BIT_LENGTH_SHIFT) + waker_page_offset
     }
 
-    pub fn has_completed(&self, task_id: u64) -> Option<bool> {
-        let pin_slab_index: usize = match self.task_ids.get(&task_id) {
-            Some(pin_slab_index) => *pin_slab_index,
+    pub fn has_completed(&self, task_id: TaskId) -> Option<bool> {
+        let pin_slab_index: InternalId = match self.task_ids.get(&task_id) {
+            Some(pin_slab_index) => pin_slab_index,
             None => return None,
         };
+
+        // Add a check here for whether the index is valid.
```

Comment:
Should this be a TODO comment?

Reply:
Sorry this was a lingering comment from before. I have moved it now